torch/_dynamo/convert_frame.py

```diff
@@ -1000,7 +1000,10 @@ def get_traced_fn(mod: Any) -> tuple[FunctionType, Optional[object]]:
     import inspect
 
     if isinstance(mod, torch.nn.Module):
-        mod = mod.forward
+        if len(mod._forward_pre_hooks) == 0 and len(mod._forward_hooks) == 0:
+            mod = mod.forward
+        else:
+            mod = mod.__call__
```

Comment:
why can't we just use mod.__call__ instead?

Reply:
@tugsbayasgalan it's an optimization for debugging, because tracing `__call__` will make the signature to be `(*args, **kwargs)` and usually `forward()` will have more user friendly arg names.